pygments/lexers/gleam.py

```diff
@@ -0,0 +1,74 @@
+"""
+    pygments.lexers.gleam
+    ~~~~~~~~~~~~~~~~~~~~~
+
+    Lexer for the Gleam programming language.
+
+    :copyright: Copyright 2006-2024 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from pygments.lexer import RegexLexer, words, bygroups
+from pygments.token import Comment, Operator, Keyword, Name, String, \
+    Number, Punctuation, Whitespace
+
+__all__ = ['GleamLexer']
+
+
+class GleamLexer(RegexLexer):
+    """
+    Lexer for the Gleam programming language (version 1.0.0).
+    """
+
+    name = 'Gleam'
+    url = 'https://gleam.run/'
+    filenames = ['*.gleam']
+    aliases = ['gleam']
+    mimetypes = ['text/x-gleam']
+    version_added = '2.17'
+
+    keywords = words((
+        'as', 'assert', 'auto', 'case', 'const', 'delegate', 'derive', 'echo',
+        'else', 'fn', 'if', 'implement', 'import', 'let', 'macro', 'opaque',
+        'panic', 'pub', 'test', 'todo', 'type', 'use',
+    ), suffix=r'\b')
+
+    tokens = {
+        'root': [
+            # Comments
+            (r'(///.*?)(\n)', bygroups(String.Doc, Whitespace)),
+            (r'(//.*?)(\n)', bygroups(Comment.Single, Whitespace)),
+
+            # Keywords
+            (keywords, Keyword),
+            (r'([a-zA-Z_]+)(\.)', bygroups(Keyword, Punctuation)),
+
+            # Punctuation
+            (r'[()\[\]{}:;,@]', Punctuation),
```

Comment:
Add a trailing `+` so you can match more than one punctuation character.

Reply:
Added, thanks